### PR TITLE
[#12656] Student Home Page: Indicate default sort 

### LINKED ISSUE #12656: Student Home Page: Indicate default sort
The default sort for the courses on student home page seems to be by course ID; however, there's no indication of this via the sort by buttons:

Screenshot 2023-12-12 at 7 36 06 PM

In contrast, the default sort (by creation date) is indicated in the instructor home page:

Screenshot 2023-12-12 at 7 36 47 PM

My suggestion would be to:
- Have the default sort on the student home page be by creation date as well - this seems to be more meaningful than course ID
- Indicate the default sort on the student home page

### PR DESCRIPTION
**Pull request by averheecke-scottlogic**
Fixes #12656

**Outline of Solution**
The problem was that the default sortBy was set to NONE. Changed this to COURSE_CREATION_DATE to result in the desired functionality.

### CODE REVIEW

**Comment by domlimm** [comment]
Hello @averheecke-scottlogic, please fix the failing snapshot test before we proceed to review your PR. You may update it by running `npm run test` in the root directory of the project. You may find more details on snapshot testing [here](https://teammates.github.io/teammates/snapshot-testing.html).

**Comment by weiquu** [comment]
Hi @averheecke-scottlogic, thanks for the PR! The button now shows that the default sort is by creation date, however I don't think that that is the actual behaviour.

Here is my student home page immediately after loading (I've closed the panels for clarity):
Screenshot 2023-12-17 at 1 31 34 AM

Upon clicking on the "Course ID" button, the sort remains exactly the same. When I click back onto the "Creation Date" button, the sort becomes as such:
Screenshot 2023-12-17 at 1 31 48 AM

It appears that even though the button indicates that the default sort is by creation date, the sort is still by Course ID. Could you please take a look?

**Comment by averheecke-scottlogic** [comment]
Hi @weiquu, the correct desired functionality should now be added :)

File: src/web/app/pages-student/student-home-page/student-home-page.component.ts [hunk 1]
```diff
@@ -114,7 +114,8 @@ export class StudentHomePageComponent implements OnInit {
             });
           });
 
-          this.courses.sort((a: StudentCourse, b: StudentCourse) => (a.course.courseId > b.course.courseId ? 1 : -1));
+          this.courses.sort((a: StudentCourse, b: StudentCourse) =>
+            (a.course.creationTimestamp < b.course.creationTimestamp ? 1 : -1));
           this.courses.slice(0, 3).forEach((course: StudentCourse) => {
             course.isTabExpanded = true;
             this.loadFeedbackSessionsForCourse(course.course.courseId);
```
Comment (weiquu) [inline src/web/app/pages-student/student-home-page/student-home-page.component.ts:121]: Would suggest replacing this with `this.sortCoursesBy(this.sortBy)`, the rationale being that there is already a function that does this functionality, and also this change makes it easier for others to see at a high level what is going on (i.e. that we are sorting courses by the current/default sort by)

**Review by weiquu — CHANGES_REQUESTED** [review]
Looks good! Just 1 small change

File: src/web/app/pages-student/student-home-page/student-home-page.component.ts (hunk 1 above)
Comment (averheecke-scottlogic) [reply src/web/app/pages-student/student-home-page/student-home-page.component.ts]: I have implemented the change, hope that's what you meant :)

Comment (weiquu) [reply src/web/app/pages-student/student-home-page/student-home-page.component.ts]: Yes that works, thanks!

**Review by weiquu — APPROVED** [review]
LGTM

**Review by domlimm — APPROVED** [review]
LGTM, thank you for your contribution!

**Pull request merged**